spec/requests/api/querying_spec.rb

```diff
@@ -457,6 +457,42 @@ def create_vms_by_name(names)
       expect(response.parsed_body).to include(expected)
       expect(response).to have_http_status(:ok)
     end
+
+    it "supports filtering by compressed id" do
+      vm1, _vm2 = create_vms_by_name(%w(aa bb))
+
+      run_get vms_url, :expand => "resources",
+                       :filter => ["id = #{ApplicationRecord.compress_id(vm1.id)}"]
+
+      expect_query_result(:vms, 1, 2)
+      expect_result_resources_to_match_hash([{"name" => vm1.name, "guid" => vm1.guid}])
+    end
+
+    it "supports filtering by compressed id as string" do
+      _vm1, vm2 = create_vms_by_name(%w(aa bb))
+
+      run_get vms_url, :expand => "resources",
+                       :filter => ["id = '#{ApplicationRecord.compress_id(vm2.id)}'"]
+
+      expect_query_result(:vms, 1, 2)
+      expect_result_resources_to_match_hash([{"name" => vm2.name, "guid" => vm2.guid}])
+    end
+
+    it "supports filtering by compressed id on *_id named attributes" do
+      zone = FactoryGirl.create(:zone, :name => "api_zone")
+      ems1 = FactoryGirl.create(:ems_vmware, :zone => zone)
+      ems2 = FactoryGirl.create(:ems_vmware, :zone => zone)
+      host = FactoryGirl.create(:host)
+
+      _vm = FactoryGirl.create(:vm_vmware, :host => host, :ems_id => ems1.id, :raw_power_state => "poweredOn")
```

Comment:
I believe setting association ids explicitly can cause some callbacks to fail..... does `:ext_management_system => ems1` work?